src/check/parse.zig

```diff
@@ -46,5 +44,53 @@ pub fn parse(env: *base.ModuleEnv, source: []const u8) IR {
         .tokens = result.tokens,
         .store = parser.store,
         .errors = errors,
+        .root_node_idx = idx,
     };
 }
+
+/// Parses a single Roc file.  The returned AST should be deallocated by calling deinit
+/// after its data is used to create the next IR, or at the end of any test.
+pub fn parse(env: *base.ModuleEnv, source: []const u8) IR {
+    return runParse(env, source, parseFileAndReturnIdx);
+}
+
+fn parseFileAndReturnIdx(parser: *Parser) u32 {
+    parser.parseFile();
+    return 0;
+}
+
+fn parseExprAndReturnIdx(parser: *Parser) u32 {
+    const id = parser.parseExpr();
```

Comment:
Based on the comment below:

> The returned AST should be deallocated by calling deinit

Shouldn't there be a deinit call here of some sort. Same for the other functions of this form in the file.

Reply:
Not here.  That's the responsibility of the caller (actually the caller of parse* functions) - that's what the comment on those functions is for.  That AST is going to be used for a number of purposes past calling this.

The deinit that is relevant is deferred on line 446 of snapshot.zig in the updated code.